packages/react-reconciler/src/__tests__/ReactIncrementalScheduling-test.js

```diff
@@ -287,12 +287,12 @@ describe('ReactIncrementalScheduling', () => {
         instance.setState({tick: 2});
       });
 
-      // TODO: why does this flush sync?
       expect(Scheduler).toFlushAndYieldThrough([
         'render: 2',
         'componentDidUpdate: 2',
         'componentDidUpdate (before setState): 2',
         'componentDidUpdate (after setState): 2',
+        // This renders because the scheduled update is default, which is sync.
```

Comment:
This isn't right, the "3" update is deferred (`deferredUpdates` is basically just `startTransition`, we should remove it)

The reason it appears in this assertion is that `toFlushAndYieldThrough` will keep rendering until the logs match, and only fails if the expected items are different or if it's unable to yield after the last item.

You should be able to remove the `gate` condition on line 285. if you wrap the `setState({tick: 2})` in `startTransition`, i.e. this behavior should work in both branches:

```js
    // Increment the tick to 2. This will trigger an update inside cDU. Flush
    // the first update without flushing the second one.
    React.startTransition(() => {
      instance.setState({tick: 2});
    });

    expect(Scheduler).toFlushAndYieldThrough([
      'render: 2',
      'componentDidUpdate: 2',
      'componentDidUpdate (before setState): 2',
      'componentDidUpdate (after setState): 2',
    ]);
    expect(ReactNoop).toMatchRenderedOutput(<span prop={2} />);

    // Now flush the cDU update.
    expect(Scheduler).toFlushAndYield(['render: 3', 'componentDidUpdate: 3']);
    expect(ReactNoop).toMatchRenderedOutput(<span prop={3} />);
```

Reply:
Could also use `toFlushUntilNextPaint` instead

Reply:
Ah, right nice catch.